### PR TITLE
Add Workers

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
Add workers deployment support in actuators

### CODE REVIEW

**Comment by greut** [comment]
[CH445]

**Comment by greut** [comment]
woot :)

> 0213 12:13:01.108730   25640 clusterdeployer.go:160] Done provisioning cluster. You can now access your cluster with kubectl --kubeconfig kubeconfig
I0213 12:13:01.108894   25640 createbootstrapcluster.go:36] Cleaning up bootstrap cluster.

**Comment by greut** [comment]
almost there.

> W0213 14:01:07.627730       1 controller.go:186] unable to create machine my-exoscale-node-62cg4: failed to obtain token for node "my-exoscale-node-62cg4" to join cluster "my-exoscale-cluster": failed to retrieve kubeconfig for cluster "my-exoscale-cluster": Provisionner exoscale GetKubeConfig() failed to run ssh cmd: unable to connect to cluster instance: dial tcp: lookup <nil>: no such host

File: pkg/cloud/exoscale/actuators/cluster/actuator.go [hunk 1]
```diff
@@ -328,12 +328,20 @@ func (*Actuator) GetIP(cluster *clusterv1.Cluster, machine *clusterv1.Machine) (
...
+	if !ok {
+		return "", fmt.Errorf("failed to get user in master machine spec: %q", master.Name)
+	}
+	password, ok := master.Annotations[exoscalev1.ExoscalePasswordAnnotationKey]
+	if !ok {
+		return "", fmt.Errorf("failed to get password in master machine spec: %q", master.Name)
 	}
 
-	sshclient := exossh.NewSSHClient(machineStatus.IP.String(), machineStatus.User, machineStatus.Password)
+	sshclient := exossh.NewSSHClient(masterIP, user, password)
```
Comment (greut) [inline pkg/cloud/exoscale/actuators/cluster/actuator.go]: good call :)

Comment (pierre-emmanuelJ) [reply pkg/cloud/exoscale/actuators/cluster/actuator.go]: yes good call, thanks for the refactoring of this function!

Comment (pierre-emmanuelJ) [reply pkg/cloud/exoscale/actuators/cluster/actuator.go]: why not putting sshkey in annotation to continue to use it instead of `user` and `password`?

Comment (greut) [reply pkg/cloud/exoscale/actuators/cluster/actuator.go]: pure laziness

File: pkg/cloud/exoscale/actuators/machine/provisioning.go [hunk 2]
```diff
@@ -296,7 +296,7 @@ func (a *Actuator) getControlPlaneMachine(machine *clusterv1.Machine) (*clusterv
 	controlPlaneList := a.getControlPlaneMachines(machineList)
 
 	//XXX work only with 1 master at the moment
-	if len(controlPlaneList) != 1 {
+	if len(controlPlaneList) == 1 {
```
Comment (pierre-emmanuelJ) [inline pkg/cloud/exoscale/actuators/machine/provisioning.go]: Thanks 🎈

**Comment by greut** [comment]
:100:

screenshot from 2019-02-15 10-18-20


```
% kubectl --kubeconfig kubeconfig get machines
                                                                                       
NAME                       AGE
my-exoscale-master-dt5pn   4m
my-exoscale-node-vhngq     4m

% kubectl --kubeconfig kubeconfig get nodes   
NAME                       STATUS   ROLES    AGE     VERSION                           
my-exoscale-master-dt5pn   Ready    master   4m54s   v1.12.5                           
my-exoscale-node-vhngq     Ready    <none>   45s     v1.12.5
```

**Pull request merged**